source/agora/node/Ledger.d

```diff
@@ -119,37 +119,71 @@ public class Ledger
             throw new Exception("Genesis block loaded from disk is " ~
                 "different from the one in the config file");
 
-        // need to regenerate the UTXO set, starting from the genesis block
+        // need to rebuild the UTXO set and Validator set,
+        // starting from the genesis block
         if (this.utxo_set.length == 0)
         {
+            ManagedDatabase.beginBatch();
+            scope (failure) ManagedDatabase.rollback();
+
+            // clear validator set
+            this.enroll_man.removeAllValidators();
+
             Block block;
+            Block last_read_block = gen_block;
             foreach (height; 0 .. this.last_block.header.height + 1)
             {
                 this.storage.readBlock(block, Height(height));
```

Comment:
@TrustHenry can you move `Block block;` inside here please?

Reply:
Yes, ~